src/UseCases/GenerateBankData/GenerateBankDataFromGermanLegacyBankDataUseCase.php

```diff
@@ -2,27 +2,23 @@
 
 declare( strict_types = 1 );
 
-namespace WMDE\Fundraising\PaymentContext\UseCases\GenerateIban;
+namespace WMDE\Fundraising\PaymentContext\UseCases\GenerateBankData;
 
 use WMDE\Fundraising\PaymentContext\Domain\BankDataGenerator;
 use WMDE\Fundraising\PaymentContext\Domain\IbanBlocklist;
-use WMDE\Fundraising\PaymentContext\ResponseModel\IbanResponse;
+use WMDE\Fundraising\PaymentContext\UseCases\IbanResponse;
 
-/**
- * @license GPL-2.0-or-later
- * @author Kai Nissen <kai.nissen@wikimedia.de>
- */
-class GenerateIbanUseCase {
+class GenerateBankDataFromGermanLegacyBankDataUseCase {
 
-	private $bankDataGenerator;
-	private $ibanBlocklist;
+	private BankDataGenerator $bankDataGenerator;
+	private IbanBlocklist $ibanBlocklist;
 
 	public function __construct( BankDataGenerator $bankDataGenerator, IbanBlocklist $ibanBlocklist ) {
 		$this->bankDataGenerator = $bankDataGenerator;
 		$this->ibanBlocklist = $ibanBlocklist;
 	}
 
-	public function generateIban( GenerateIbanRequest $request ): IbanResponse {
+	public function generateIban( GenerateIbanFromGermanLegacyBankDataRequest $request ): IbanResponse {
```

Comment:
We don't need a request object here, bank account and bank code as strings will make this method more readable and the code shorter